libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -75,8 +79,13 @@ size_t ContractCompiler::compileConstructor(
 )
 {
 	CompilerContext::LocationSetter locationSetter(m_context, _contract);
-	initializeContext(_contract, _contracts);
-	return packIntoContractCreator(_contract);
+	if (_contract.isLibrary())
```

Comment:
This change of not running the constructor code for libraries together with the assertions could be merged separately as it is just a codegen reinforcement of the typechecker decisions.